kura/org.eclipse.kura.driver.ble.xdk/src/main/java/org/eclipse/kura/internal/driver/ble/xdk/XdkDriver.java

```diff
@@ -0,0 +1,551 @@
+/**
+ * Copyright (c) 2019 Eurotech and/or its affiliates and others
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Eurotech
+ */
+package org.eclipse.kura.internal.driver.ble.xdk;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNull;
+import static org.eclipse.kura.channel.ChannelFlag.FAILURE;
+import static org.eclipse.kura.channel.ChannelFlag.SUCCESS;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import org.eclipse.kura.KuraBluetoothIOException;
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.bluetooth.le.BluetoothLeAdapter;
+import org.eclipse.kura.bluetooth.le.BluetoothLeDevice;
+import org.eclipse.kura.bluetooth.le.BluetoothLeService;
+import org.eclipse.kura.channel.ChannelFlag;
+import org.eclipse.kura.channel.ChannelRecord;
+import org.eclipse.kura.channel.ChannelStatus;
+import org.eclipse.kura.channel.listener.ChannelListener;
+import org.eclipse.kura.configuration.ConfigurableComponent;
+import org.eclipse.kura.driver.ChannelDescriptor;
+import org.eclipse.kura.driver.Driver;
+import org.eclipse.kura.driver.PreparedRead;
+import org.eclipse.kura.type.DataType;
+import org.eclipse.kura.type.TypedValue;
+import org.eclipse.kura.type.TypedValues;
+import org.eclipse.kura.util.base.TypeUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class XdkDriver implements Driver, ConfigurableComponent {
+
+    private static final Logger logger = LoggerFactory.getLogger(XdkDriver.class);
+
+    private static final int TIMEOUT = 5;
+    private static final String INTERRUPTED_EX = "Interrupted Exception";
+
+    private static final byte m1 = 0x01;
```

Comment:
Please consider renaming it as "message1" or "message1Index" or something that describes what this and m2 are